pkg/kubelet/kubelet.go

```diff
@@ -2010,6 +2011,12 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 
 	// Wait for volumes to attach/mount
 	if err := kl.volumeManager.WaitForAttachAndMount(ctx, pod); err != nil {
+		var volumeAttachLimitErr *volumemanager.VolumeAttachLimitExceededError
```

Comment:
Continuing on this comment: https://github.com/kubernetes/kubernetes/pull/132933#discussion_r2211805558

It is not trivial to compare everything got prepared by that line and what and how being cleaned up later. The only worrisome thing I found is the DRA - it is prepared after this line, but unprepared all the time. So DRA plugin will see the Unprepare call for the resources that was never prepared.

This is similar to a pod entering crash loop backoff because of CSI failure and later being deleted. So DRA must be ready for this.

That said, from looking over all codepaths I can think of - this looks OK.

Reply:
@pohly FYI

Reply:
I think this should be ok but we should make sure it's documented for driver authors.